packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -362,47 +368,45 @@ export const _generateStyles = (
 
     let newStyles: Record<string, CSSProperties> = {};
     if (kind === "primary") {
-        const boxShadowInnerColor: string = light
-            ? theme.color.bg.primary.inverse
-            : theme.color.bg.primary.default;
+        const focusStyling = {
+            outlineColor: light ? theme.color.bg.primary.default : color,
+            outlineOffset: 2,
+            outlineStyle: "solid",
+            outlineWidth: 2,
+        };
+
+        const activePressedStyling = {
+            background: light ? fadedColor : activeColor,
+            outlineColor: light ? fadedColor : activeColor,
+        };
 
         newStyles = {
             default: {
                 background: light ? theme.color.bg.primary.default : color,
                 color: light ? color : theme.color.text.inverse,
                 paddingLeft: padding,
                 paddingRight: padding,
+                // TODO(WB-1844): Change this when we get final designs for
+                // hover.
+                [":hover:not([aria-disabled=true])" as any]: focusStyling,
+                [":focus-visible:not([aria-disabled=true])" as any]:
+                    focusStyling,
+                [":active:not([aria-disabled=true])" as any]:
+                    activePressedStyling,
             },
-            focus: {
-                // This assumes a background of white for the regular button and
-                // a background of darkBlue for the light version. The inner
-                // box shadow/ring is also small enough for a slight variation
-                // in the background color not to matter too much.
-                boxShadow: `0 0 0 1px ${boxShadowInnerColor}, 0 0 0 3px ${
-                    light ? theme.color.bg.primary.default : color
-                }`,
```

Comment:
**note:** Took the opportunity to switch from `boxShadow` to `outline`. This could help solving this ticket: https://khanacademy.atlassian.net/browse/WB-1712